src/main/java/gov/nasa/pds/objectAccess/TableReader.java

```diff
@@ -325,14 +360,11 @@ public long getRecordSize(URL dataFile, Object table) throws Exception {
 			throw ex;
 		}
 		if (adapter instanceof TableDelimitedAdapter) {			
-			// chek this again
-			is.skip(offset);
-			bufferedReader = new BufferedReader(new InputStreamReader(is, "US-ASCII"));
-			int numLines = 0;
-			while (bufferedReader.readLine()!=null) {
-			   ++numLines;
-			}
-			this.recordSize = numLines;
+            // Removed old usage of BufferedReader to count lines which is problematic when file is large.
+            // Use new function countCarriageReturn() to count number of lines of any file sizes.
+
+            this.recordSize = this.countCarriageReturn(dataFile);
```

Comment:
@qchaupds per our convo today and [this ticket](https://github.com/NASA-PDS/validate/issues/325), I don't think this method will work in the event the CRLF is embedded within a record. we should consider checking for record count using opencsv when we are already reading in the data to do content validation.

Reply:
from @qchaupds The code change in TableReader.java in pds4-jparser now uses the record length to calculate the number of records.  No need to read the file.